### DISCUSSION TITLE
Have a doubt following the breakout tutorial from getting started

### DISCUSSION THREAD

**Discussion by Charkui** [excaliburjs/Excalibur#1959] [Q&A]
Hi all!
My doubt is this: It's there a way that an Fixed-collision actor doesn't collide with a passive one? I was following the getting-started tutorial but... from the excitement for learn how to create videogames, I decided to jump directly to use the wepack template, that defines a Game class that extends Engine, I was following the instructions and trying to make it modular at the same time. But when I finished the ball Actor segment from the tutorial. The ball wasn't colliding with the paddle.
The files are the next:
paddle.ts
`import {Actor, CollisionType, Color} from 'excalibur'

export default class Paddle extends Actor {
    constructor(gameHeight: number){
        super({
            x: 150,
            y: gameHeight - 40,
            width: 200,
            height: 20,
            color: Color.Chartreuse
          })
          this.body.collider.type = CollisionType.Fixed;
    }
}
`

ball.ts
`import {Actor, CollisionType, Color} from 'excalibur'

export default class Ball extends Actor {
    constructor(posX: number, posY: number){
        super({
            x: posX,
            y: posY,
            color: Color.Red
        })

        this.body.useCircleCollider(10);
        this.body.collider.type = CollisionType.Passive;
    }
}`

game.ts
`import { Engine, Loader, DisplayMode, vec } from 'excalibur';
import { LevelOne } from './scenes/level-one/level-one';
import Paddle from './actors/player/paddle';
import Ball from './actors/ball/ball';
import { Resources } from './resources';

/**
 * Managed game class
 */
class Game extends Engine {
  private paddle: Paddle;
  private ball: Ball;
  private levelOne: LevelOne;

  constructor() {
    super({ displayMode: DisplayMode.FullScreen });
  }

  public start() {

    // Create new scene with a player
    this.levelOne = new LevelOne(this);
    this.add('levelOne', this.levelOne);
    this.paddle = new Paddle(this.drawHeight);
    this.ball = new Ball(100,300);
    this.configureBall();
    setTimeout(this.setBallVelocity.bind(this), 1000);
    this.levelOne.add(this.paddle);
    this.levelOne.add(this.ball);

    this.input.pointers.primary.on('move', evt => {
      this.paddle.pos.x = evt.worldPos.x;
    })
    // Automatically load all default resources
    const loader = new Loader(Object.values(Resources));
    return super.start(loader);
  }

  private configureBall() {
    this.ball.on('postupdate', () => {
      if(this.ball.pos.x < this.ball.width / 2){
        this.ball.vel.x *= -1;
      }

      if(this.ball.pos.x + this.ball.width / 2 > this.drawWidth){
        this.ball.vel.x *= -1;
      }

      if(this.ball.pos.y < this.ball.height / 2){
        this.ball.vel.y *= -1;
      }
    })
  }

  private setBallVelocity(){
    this.ball.vel = vec(100,100);
  }
}

const game = new Game();
game.start().then(() => {
  game.goToScene('levelOne');
});
`

level-one.ts
`import { Engine, Scene } from 'excalibur';

/**
 * Managed scene
 */
export class LevelOne extends Scene {
  public onInitialize(engine: Engine) {}
  public onActivate() {}
  public onDeactivate() {}
}
`
Thanks for your time, I'm really enjoying excalibur and the tutorial!

**Comment by eonarheim** [comment]
Hi @Charkui I have the full sample posted [here](https://github.com/excaliburjs/sample-breakout/blob/main/src/main.ts) to compare.

With the `Passive` collision type on the ball it is important to [handle the collision event](https://github.com/excaliburjs/sample-breakout/blob/main/src/main.ts#L137) to make the ball bounce, otherwise it will pass right through the paddle.

Hope this helps,

Erik
